clients/admin-ui/src/features/consent-reporting/ConsentReporting.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  Button,
+  HStack,
+  Input,
+  InputGroup,
+  InputLeftAddon,
+  useToast,
+} from "@fidesui/react";
+import { useState } from "react";
+
+import { getErrorMessage } from "~/features/common/helpers";
+import { useLazyDownloadReportQuery } from "~/features/consent-reporting/consent-reporting.slice";
+
+const ConsentReporting = () => {
+  const [startDate, setStartDate] = useState<string>("");
+  const [endDate, setEndDate] = useState<string>("");
+
+  const toast = useToast();
+
+  const [downloadReportTrigger, { isLoading }] = useLazyDownloadReportQuery();
+
+  const handleDownloadClicked = async () => {
+    const result = await downloadReportTrigger({ startDate, endDate });
+    if (result.isError) {
+      const message = getErrorMessage(
+        result.error,
+        "A problem occurred while generating your consent report.  Please try again."
+      );
+      toast({ status: "error", description: message });
+    } else {
```

Comment:
The actual download had to be moved out of the slice file to be able to handle errors.